File: hackerrank/practice/minimum_distances.py
#!/usr/bin/env python

def minimum_distance(test_array):
    
    distances = []
    for index, value in enumerate(test_array):
        for index2, value2 in enumerate(test_array):
            if value == value2 and index != index2:
                distance = index2 - index
                if distance < 0:
                    distance = distance * -1
                distances.append(distance)
    if len(distances) > 0:
        print(min(distances))
    else:
        print(-1)
            

for _ in range(int(input())):
    n = int(input())
    input_array = list(map(int, input().split()))
    minimum_distance(input_array)
            

File: hackerrank/practice_cpp/minimum_distances.cpp
#include<iostream>

int minimum_distance(int arr[100000], int n);

int minimum_distance(int arr[100000], int n)
{
    int distance;
    int minimum_distance;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            if(arr[i] == arr[j] && i != j)
            {
                distance = abs(j - i);
                minimum_distance = distance;
                if (distance < minimum_distance)
                {
                    minimum_distance = distance;
                }
            }
        }   
    }
    return minimum_distance;
}

int main()
{
    int test_array[100000];
    int n;
    std :: cin >> n;
    for (int i = 0; i < n; i++)
    {
        std :: cin >> test_array[i];
    }
    
    std :: cout << minimum_distance(test_array, n) << std :: endl;

    return 0;
}
